### PR TITLE
Hard coded minimum width for components

### LINKED ISSUE #18: When resizing sliders/graph components, they can become too small and unresizable
Potential fix: minimum element width/height so that user cannot resize the element to the point where it is so thin/flat that it cannot be resized to a larger width/height.

### PR DESCRIPTION
**Pull request by cchan987**
hard coded a minimum width and height for components so that they don't get lost if resized to a size that is too small to click on.

fixes #18

### CODE REVIEW

File: nengo_viz/static/viz.js [hunk 1]
```diff
@@ -70,6 +70,12 @@ VIZ.Component = function(args) {
             var target = event.target;
             var newWidth = parseFloat(target.style.width) + event.dx;
             var newHeight = parseFloat(target.style.height) + event.dy;
+            if (newWidth < 30){
+                newWidth = 30;
+            }
+            if (newHeight < 50){
+                newHeight = 50;
+            }
```
Comment (tcstewar) [inline nengo_viz/static/viz.js]: Cool!  But let's make the minimum size an attribute of the class:

```js
            if (newWidth < self.minWidth) {
                newWidth = self.minWidth;
            }
            if (newHeight < self.minHeight) {
                newHeight = self.minHeight;
            }
```

Comment (tcstewar) [inline nengo_viz/static/viz.js]: (and, of course, add ```this.minHeight = 50;``` to the constructor)

**Comment by cchan987** [comment]
Sliders now show their scaled value.

Sliders cannot be moved unless mouse is over them.

fixes #12

**Comment by cchan987** [comment]
mouse over requirement may affect touch compatibility, but I'm not sure...

**Comment by cchan987** [comment]
Made the onmouseleave have a timeout to support users moving very slider very rapidly

**Comment by tcstewar** [comment]
@cchan987 , it would help a lot to have these different independent changes in different branches/pull requests.  This Pull Request was opened for doing things with the minimum width of components, which doesn't have much to do with the slider stuff you're working on now.

It also simplifies things a lot for me testing the changes if they're done in a branch of the original repository.  The general flow is something like this:

 - Decide on a new feature to work on
 - Assign that issue to yourself
 - Start a new branch ```git checkout -b improved-sliders```
 - Work on the code till you're happy with it
 - Commit the code changes ```git add my_changed_file1.py``` ```git commit```
 - Push the code changes to github ```git push origin improved-sliders```
 - Make the pull request on github
 - Go back to the master branch ```git checkout master```
 - Go through this loop again.

It definitely takes some getting used to this style of doing things....  :)  But it's nice to have things in separate branches so I could easily accept the minwidth stuff while we sort out the slider stuff.  :)

**Comment by cchan987** [comment]
sure thing, I'll move the other changes to a different branch

**Pull request merged**